src/components/common/graph/d3-graph-elevation.vue

```diff
@@ -0,0 +1,127 @@
+  // TODO:
+  // if (this.measureTooltipElement_  !== null) {
+  //   this.measureTooltipElement_.parentNode.removeChild(
+  //       this.measureTooltipElement_);
+  //   this.measureTooltipElement_ = null;
+  //   this.measureTooltip_ = null;
+  // }
+}
+
+// function createMeasureTooltip() {
+// TODO:
+// this.removeMeasureTooltip_();
+// this.measureTooltipElement_ = document.createElement('DIV');
+// this.measureTooltipElement_.classList.add('tooltip');
+// this.measureTooltipElement_.classList.add('ngeo-tooltip-measure');
+// this.measureTooltip_ = new olOverlay({
+//   element: this.measureTooltipElement_,
+//   offset: [0, -15],
+//   positioning: 'bottom-center'
+// });
+// this.map_.addOverlay(this.measureTooltip_);
+// }
+
+let profile: Profile, elevationData: any, poiData: any
+let selection: Selection<any, unknown, null, undefined>
+let hoverCallback = () =>
+  // point: any,
+  // dist: number,
+  // xUnits: string,
+  // elevation: Elevations,
+  // yUnits: string
+  {
+    // TODO:
+    if (props.showTooltip) {
+      // An item in the list of points given to the profile.
+      // this['point'] = point;
+      // this.featureOverlay_.clear();
+      // const curPoint = new olGeomPoint([point['x'], point['y']]);
+      // curPoint.transform('EPSG:2169', this.map_.getView().getProjection());
+      // var positionFeature = new olFeature({
+      //   geometry: curPoint
+      // });
+      // this.featureOverlay_.addFeature(positionFeature);
+      // createMeasureTooltip();
+      // this.measureTooltipElement_.innerHTML = this.distanceLabel_ +
+      //     this.formatDistance_(dist, xUnits) +
+      //     '<br>' +
+      //     this.elevationLabel_ +
+      //     this.formatElevation_(elevation['line1'], yUnits);
+      // this.measureTooltip_.setPosition(curPoint.getCoordinates());
+      // this.snappedPoint_.setGeometry(new olGeomPoint([point.x, point.y]));
+    }
+  }
+let profileOptions: ProfileOptions = {
+  linesConfiguration: {
+    line1: {
+      zExtractor: item =>
+        'values' in item && 'dhm' in item['values']
+          ? parseFloat(item['values']['dhm'].toPrecision(5))
+          : 0,
+    },
+  },
+  distanceExtractor: item => ('dist' in item ? item['dist'] : 0),
+  hoverCallback,
+  outCallback: () => {
+    // TODO:
+    if (props.showTooltip) {
+      // this['point'] = null;
+      removeMeasureTooltip()
+      // this.featureOverlay_.clear();
+      // this.snappedPoint_.setGeometry(null);
+    }
+  },
+  formatter: {
+    xhover: (dist, units) => parseFloat(dist.toPrecision(3)) + ' ' + units,
+    yhover: (elevation, units) =>
+      parseFloat(elevation.toPrecision(4)) + ' ' + units,
+  },
+}
+
+onMounted(() => {
+  selection = d3.select(graph.value)
+  profile = D3GraphElevation(profileOptions)
+
+  refreshData()
+})
+
+olEvents.listen(window, 'resize', debounce(refreshData, 50))
+
+function refreshData() {
+  if (profile !== undefined) {
+    selection.datum(dataset).call(profile)
```

Comment:
this is a very obscure use of the d3 library.
It worked in v3, it works in v4, but I don't understand it.

Reply:
is it d3 that is obscure for you? or is it the way we use it?